src/init.rs

```diff
@@ -158,6 +158,12 @@ if [[ ${preexec_functions[(ie)starship_preexec]} -gt ${#preexec_functions} ]]; t
     preexec_functions+=(starship_preexec);
 fi;
 STARSHIP_START_TIME="$(date +%s)";
+function zle-keymap-select
+{
+    PROMPT=$(starship prompt --keymap=$KEYMAP)
+    zle reset-prompt
+}
+zle -N zle-keymap-select
```

Comment:
Just so that I understand this correctly: if I'm using emacs mode, or vi mode without ever switching into normal mode, this function will never fire right? It'll be just like using starship before this change?

Or is there some situation in which the prompt might redraw in spite of no mode change occuring?